services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,62 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if(!event){
+    return ERR_CODE_INVALID_ARG;
+  }
+
+
+  if (!thermalMgrQueueHandle) {
+    return ERR_CODE_INVALID_QUEUE_MSG;
+  }
 
+  if(xQueueSend(thermalMgrQueueHandle, (const void *) event, 0) != pdTRUE){
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t eventInterrupt;
+  eventInterrupt.type = THERMAL_MGR_EVENT_INTERRUPT;
+  thermalMgrSendEvent(&eventInterrupt);
+
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  thermal_mgr_event_t event;
+  float temp;
+  error_code_t errCode;
   while (1) {
-    
+    if(xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdTRUE){
+
+      if(event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD ){
+        errCode = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
+        if(errCode != ERR_CODE_SUCCESS){
+          break;
```

Comment:
just continue if there is an error, you should not be breaking out of the while(1) loop